### ISSUE TITLE
Example of an old score file that causes a crash in MS 4 when opening and in MS 2 if corrupted measure is interacted with...

### ISSUE THREAD

**Issue by mtaalas** [musescore/MuseScore#25031]
### Issue type

Crash or freeze

### Description with steps to reproduce

I've attached three files, the original and one that has the corrupted measure removed with text editor AND one that I had removed the corruption and then saved with Musescore 2.01 again, that didn't resolve the MS 4 crash issue.

In MS 4.4.2:

1. Open the attached file in Musescore 4.4.2 [Corrupted_files.zip](https://github.com/user-attachments/files/17249506/Corrupted_files.zip)
2. Get warning about corrupted measure
3. Choose to open anyway
4. Select to either replace fonts or not, but try to open the file by pressing ok.
5. Crash to desktop.

In MS 2.0.1:

1. Open attached file
2. Get a error stating that file is corrupted
3. Click "Ignore"
4. Score opens, and even plays...
5. See the last measure 125 looking weird that the MS 4 warned about.
6. Try to select the measure (to for example, delete or fix it)
7. Crash

If I go and delete the last measure using a text editor (I know XML somewhat, edited file is attached) MS 4 nor MS 2.0.1 warn about corruption anymore when opening, but MS 4 is still not able to open it but crashes similarly to 2.0.1

2.0.1 doesn't crash anymore because there's no corrupted measure to interact with to cause the crash.

Does this mean that MS 4 doesn't crash because of the corruption, but because there's something else that's incompatible with MS 4 in MS2 format?

Is the format more complex that I've understood and my edit doesn't actually fix the underlying issue?

I hope these files help narrow it down a bit and you'll figure out why at least.

### What is the latest version of MuseScore Studio where this issue is present?

OS: Windows 10 Version 2009 or later, Arch.: x86_64, MuseScore Studio version (64-bit): 4.4.2-242570931, revision: github-musescore-musescore-3130f97

### Regression

No.

### Operating system

Windows 10

### Checklist

- [X] This report follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting bugs and issues
- [X] I have verified that this issue has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar issues
- [X] I have attached all requested files and information to this report
- [X] I have attempted to identify the root problem as concisely as possible, and have used minimal reproducible examples where possible

**Comment by krasko78** [comment]
Some voltas have <anchor>0</anchor> and crash MSS. I've commented those anchors out (not the entire voltas). I don't know what their purpose is and what is lost due to this removal, but some of the voltas didn't have <anchor>0</anchor>. Hope this helps. Remove the .zip extension.
[Folkkimetallia.mscz.zip](https://github.com/user-attachments/files/17250988/Folkkimetallia.mscz.zip)

**Comment by DmitryArefiev** [comment]
Tested on Win10. This **doesn't crash** in **4.0.0-4.0.2**

**Comment by mtaalas** [comment]
> Some voltas had `<anchor>0</anchor>` and crash MSS. I've commented those anchors out (not the entire voltas). I don't know what their purpose was and what is lost due to their removal, but some of the voltas didn't have `<anchor>0</anchor>`. Hope this helps. Remove the .zip extension. [Folkkimetallia.mscz.zip](https://github.com/user-attachments/files/17250988/Folkkimetallia.mscz.zip)

Confirmed, It doesn't crash on 4.4.2 AND it plays back exactly as it should. And I see you didn't touch the measure 125, just removed the erroneous Voltas.

There are some changes to the layout in 4.4.2, but that's to be expected after 10 years and the large overhaul of the engraving system in 4.x.

So the root cause for the crash seems to have been the Voltas.

Could it be that the 2.0.1 version that I made this on back in 2014 or so (version is saved to the metadata of the original file) had a tendency to create these erroneous Voltas while I thought I'm making them properly? The score plays back and displays 1:1 as it should in 2.0.1. See this attachment PDF that I took 10 years ago of that same file and they're the same visually with 2.0.1 display.

PDF's dated 28.07.2014 by my NAS. Which should be correct.

Thank you all for checking this out and It seems I inadvertently stumbled upon a regression. :)
I'll be following how this goes and help to test this when needed.

[Folkkimetallia.pdf](https://github.com/user-attachments/files/17261485/Folkkimetallia.pdf)
